service/history/execution/state_builder.go

```diff
@@ -114,6 +114,9 @@ func (b *stateBuilderImpl) ApplyEvents(
 			return nil, err
 		}
 		versionHistories := b.mutableState.GetVersionHistories()
+		if versionHistories == nil {
+			return nil, &shared.BadRequestError{Message: "2DC workflow is not supported."}
```

Comment:
Seems too prescripted to say 2DC here. How about:
```
versionHistories is empty, which is required for NDC feature. It's probably from deprecated 2dc workflows
```

Reply:
Updated the error messages